metals/src/main/scala/scala/meta/internal/metals/FormattingProvider.scala

```diff
@@ -215,11 +218,157 @@ final class FormattingProvider(
+
+        if (ord.gt(required, default)) {
+          // check maybe they covered by fileOverride
+          if (configValues.fileOverrides.nonEmpty) {
+            val allOverriden = scalaFiles.forall(p =>
+              configValues.overrideFor(p).contains(required)
+            )
+            if (allOverriden) None else Some(required)
+          } else Some(required)
+        } else None
+      } else None
+    }
+
+    val default = configValues.runnerDialect.getOrElse(ScalafmtDialect.Scala213)
+
+    val itemsRequiresUpgrade =
+      buildTargets.sourceItemsToBuildTargets.toList.flatMap {
+        case (path, ids) =>
+          inferDialectForSourceItem(path, ids.asScala.toList, default).map(d =>
+            (path, d)
+          )
+      }
+
+    if (itemsRequiresUpgrade.nonEmpty) {
+      val (items, dialects) = itemsRequiresUpgrade.unzip
+      Some(items.map(_.toRelative(workspace)), dialects.max)
+    } else None
+  }
+
+  private def fmtVersionUpgrade(
+      configValues: ScalafmtConfig.ConfigValues
+  ): Option[String] = {
+    val major = for {
+      version <- configValues.version
+      value <- Try(version.split('.').head.toInt).toOption
+
+    } yield value
+
+    major match {
+      case Some(n) if n < 3 => Some(BuildInfo.scalafmtVersion)
+      case None => Some(BuildInfo.scalafmtVersion)
+      case _ => None
+    }
+  }
+
+  private def checkIfDialectUpgradeRequired(
+      configValues: ScalafmtConfig.ConfigValues
+  ): Future[Unit] = {
+    upgradeDialect(configValues) match {
+      case Some((directories, required))
+          if !tables.dismissedNotifications.UpdateScalafmtConf.isDismissed =>
+        val canUpdate = userConfig().scalafmtConfigPath.isEmpty
+        val params = UpdateScalafmtConf.params(directories, required, canUpdate)
+        client.showMessageRequest(params).asScala.map { item =>
+          if (item == UpdateScalafmtConf.letUpdate) {
+            val updVersion = fmtVersionUpgrade(configValues)
+
+            val text = scalafmtConf.toInputFromBuffers(buffers).text
+            val updatedText =
+              ScalafmtConfig.update(text, updVersion, Some(required))
+
+            Files.write(
+              scalafmtConf.toNIO,
+              updatedText.getBytes(StandardCharsets.UTF_8)
+            )
+          } else if (item == Messages.notNow) {
+            tables.dismissedNotifications.UpdateScalafmtConf
+              .dismiss(24, TimeUnit.HOURS)
+          } else if (item == Messages.dontShowAgain) {
+            tables.dismissedNotifications.UpdateScalafmtConf.dismissForever()
+          } else ()
+        }
+      case _ => Future.unit
+    }
+  }
+
+  def validateWorkspace(): Future[Unit] = {
+    if (scalafmtConf.exists) {
+      val text = scalafmtConf.toInputFromBuffers(buffers).text
+      ScalafmtConfig.parse(text) match {
+        case Failure(e) =>
+          scribe.error(s"Failed to parse ${scalafmtConf}", e)
+          client.showMessage(
```

Comment:
Let's not show it to the user, just log it. It's not critical, so it's fine to just have it there and it's highly possible that our parsing solution is not a 100 % valid. Especially if anyone has a more complex configuration there should be able to figure it out. Otherwise, they will keep on getting the errors every time.